### DISCUSSION TITLE
Is there a version available for ESP32-S3?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#13344] [ESP32]
I have tried using ESP-IDF 4.3, 5.0, and 5.2, but none of them were able to compile. Which version should I use?

**Comment by robert-hh** [comment]
The current version of Micropython is built with esp-idf 5.0.4.
If that fails for you, then the set-up of the idf may be wrong.

**Reply by Mars-CN** [reply]
There is no problem building ESP32 with esp-idf 5.0.4, but ESP32-S3 cannot compile properly and will report many exceptions

**Reply by robert-hh** [reply]
> There is no problem building ESP32 with esp-idf 5.0.4, but ESP32-S3 cannot compile properly and will report many exceptions

The ESP32-S3 builds fine here with esp-idf 5.0.4 and BOARD=ESP32_GENERIC_S3. The other boards build fine as well. There is something wrong with your set-up. After changing the version in the esp-idf directory, did you run `git submodule sync; git submodule update --init` in the esp-idf directory and run install.sh in that directory?

**Comment by Josverl** [comment]
Down download from:
https://micropython.org/download/ESP32_GENERIC_S3/

or if you want to build you can create/verify your setup by following the CI setup steps 
```sh
source tools/ci.sh && ci_esp32_idf_setup
```

**Reply by Mars-CN** [reply]
I need the source code, not the compiled bin file

**Comment by shariltumin** [comment]
This is the build script I use:

```sh
#!/bin/bash

export PATH="$HOME/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

echo "=========================================================="
echo "Setting up for esp32s3 build"
echo "Using esp-idf-504"
export IDF_PATH="$HOME/disk/esp/esp-idf-504"
export IDF_TARGET="esp32s3"
source $IDF_PATH/export.sh

make V=1 BOARD=ESP32S3_LOLIN_MINI_N4R2
echo "=========================================================="
```

You may need to change the paths to suit your needs. Hope this helps.
